### PR TITLE
Fix typing of ``Arguments.args``

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Based on discussion in `pylint` here: https://github.com/PyCQA/pylint/pull/5409#discussion_r762497136

I have added changelog entry to increase visibility for this change.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
👍 Justifying the necessary ``Optional`` is pretty important, because I'm sure there are a lot of "false optional" that we'll want to remove to simplify our live and have simpler typing/code.

**Comment by DanielNoord** [comment]
> 👍 Justifying the necessary `Optional` is pretty important, because I'm sure there are a lot of "false optional" that we'll want to remove to simplify our live and have simpler typing/code.

If we could redesign `astroid` I think we would have solved this differently. However, I agree with @cdce8p that in the current situation this is the best way forward. We would need to deprecate etc. otherwise..

File: astroid/nodes/node_classes.py [hunk 1]
```diff
@@ -858,8 +858,12 @@ def __init__(
         self.kwarg: Optional[str] = kwarg  # can be None
         """The name of the variable length keyword arguments."""
 
-        self.args: typing.List[AssignName]
-        """The names of the required arguments."""
+        self.args: typing.Optional[typing.List[AssignName]]
+        """The names of the required arguments.
+        Can be None if the assosciated function does not have a retrievable
+        signature and the arguments are therefore unknown. This happens with builtin
+        functions implemented in C.
```
Comment (cdce8p) [inline astroid/nodes/node_classes.py:865]:
```suggestion
        """The names of the required arguments.

        Can be None if the assosciated function does not have a retrievable
        signature and the arguments are therefore unknown.
        This happens with builtin functions implemented in C.
```

**Review by cdce8p — APPROVED** [review]
Looks good! Thanks @DanielNoord 👍🏻

**Pull request merged**